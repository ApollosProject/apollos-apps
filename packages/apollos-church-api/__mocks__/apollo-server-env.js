import fetch from 'jest-fetch-mock';

import { Constants } from 'apollos-church-api/src/connectors/rock';
import * as apolloDatasourceMocks from './apollo-datasource-mocks';
import * as rockMocks from './rock-api-mocks';

const apolloServerEnv = require.requireActual('apollo-server-env');

// eslint-disable-next-line
const Response = apolloServerEnv.Response;

const resolveWith = (data, url) =>
  Promise.resolve(
    new Response(JSON.stringify(data), {
      url,
      status: 200,
      statusText: 'OK',
      headers: new apolloServerEnv.Headers({
        'Content-Type': 'application/json',
      }),
    })
  );

fetch.mockLiveDataSourceApis = () => {
  fetch.mockImplementation((request) => {
    if (request.url.match('/api/v1/events/current')) {
      return resolveWith(apolloDatasourceMocks.liveStreamLive(), request.url);
    }
    if (request.url.match('/passage/html')) {
      return resolveWith(apolloDatasourceMocks.ESVScripture(), request.url);
    }
    return Promise.reject();
  });
};

fetch.mockRockDataSourceAPI = () => {
  fetch.mockImplementation((request) => {
    let { url } = request;
    url = decodeURI(url);
    if (!url.match(Constants.ROCK_API)) return Promise.reject();

    if (url.match('api/ContentChannels/\\d')) {
      return resolveWith(rockMocks.contentChannel());
    }

    if (url.match('api/ContentChannels')) {
      return resolveWith([rockMocks.contentChannel()]);
    }

    if (url.match('api/ContentChannelItems/test-case-no-attributes')) {
      const mock = rockMocks.contentItem();
      mock.AttributeValues = {};
      mock.Attributes = {};
      return resolveWith(mock);
    }

    if (url.match('api/ContentChannelItems/\\d')) {
      return resolveWith(rockMocks.contentItem());
    }

    if (url.match('api/ContentChannelItems')) {
      return resolveWith([rockMocks.contentItem()]);
    }

    if (url.match('api/ContentChannelItemAssociations')) {
      return resolveWith([rockMocks.contentChannelItemAssociation()]);
    }

    if (url.match('api/Auth/Login')) {
      const body = JSON.parse(request.body);
      const response = new Response('');
      if (body.Password === 'good') {
        response.headers.set('set-cookie', 'some cookie');
        return Promise.resolve(response);
      }
      return Promise.reject({ ...response, status: 401 });
    }

    if (url.match('api/People/GetCurrentPerson')) {
      return resolveWith(rockMocks.people());
    }

    if (url.match('api/People/\\d')) {
      return resolveWith(rockMocks.people());
    }

    if (url.match('api/People')) {
      if (request.method === 'POST') {
        const { Email } = JSON.parse(request.body);
        if (!Email) {
          const response = new Response('');
          response.status = 400;
          return Promise.reject(response);
        }
        return resolveWith({ personId: 35 });
      }

      return resolveWith([rockMocks.people()]);
    }

    if (url.match('api/UserLogins')) {
      if (request.method === 'POST') {
        const { UserName } = JSON.parse(request.body);
        if (!UserName) {
          const response = new Response('');
          response.status = 400;
          return Promise.reject(response);
        }
        return resolveWith({ id: 21 });
      }
      const identity = url // identity = UserName
        .split('eq')
        .pop()
        .trim(' '); // EXAMPLE URL: /api/UserLogins?$filter=UserName eq 'isaac.hardy@newspring.cc'

      if (!identity) {
        const response = new Response('');
        response.status = 400;
        return Promise.reject(response);
      }
      if (identity === `'isaac.hardy@newspring.cc'`)
        return resolveWith([rockMocks.userLogins()]);

      return resolveWith([]);
    }

    return Promise.reject();
  });
};

const apolloServerEnvMocked = {
  ...apolloServerEnv,
  Response: fetch.Response,
  fetch,
};

module.exports = apolloServerEnvMocked;
