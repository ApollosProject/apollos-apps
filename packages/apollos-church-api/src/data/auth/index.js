import { gql } from 'apollo-server';
import { createGlobalId } from '../node';

// export { default as model } from './model';
export { default as dataSource } from './data-source';

export const schema = gql`
  type AuthenticatedUser {
    id: ID!
    profile: Person
  }

  type Authentication {
    user: AuthenticatedUser
    token: String
  }

  extend type Mutation {
    authenticate(identity: String!, password: String!): Authentication
    registerPerson(email: String!, password: String!): Authentication
  }

  extend type Query {
    currentUser: AuthenticatedUser
  }
`;

export const resolver = {
  Query: {
    currentUser: (root, args, { dataSources }) =>
      dataSources.Auth.getCurrentPerson(),
  },
  AuthenticatedUser: {
    id: ({ id }, args, context, { parentType }) =>
      createGlobalId(id, parentType.name),
    profile: (authUser) => authUser,
  },
  Authentication: {
    user: (root, args, { dataSources }) => dataSources.Auth.getCurrentPerson(),
  },
  Mutation: {
    authenticate: (root, { identity, password }, { dataSources }) =>
      dataSources.Auth.authenticate({ identity, password }),
    registerPerson: (root, args, { dataSources }) =>
      dataSources.Auth.registerPerson(args),
  },
};
