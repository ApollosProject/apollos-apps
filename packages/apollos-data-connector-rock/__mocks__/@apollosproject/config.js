const ApolloServer = require.requireActual('@apollosproject/config').default;

ApolloServer.loadJs({
  ROCK_MAPPINGS: {
    CONTENT_ITEM: {
      ContentSeriesContentItem: {
        EntityType: 'ContentChannelItem',
      },
      DevotionalContentItem: {
        EntityType: 'SomeOtherEntity',
      },
      MediaContentItem: {
        EntityType: 'ContentChannelItem',
      },
      UniversalContentItem: {
        EntityType: 'ContentChannelItem',
      },
      ContentItem: {
        EntityType: 'ContentChannelItem',
      },
    },
  },
  ROCK: {
    TIMEZONE: 'America/New_York',
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
  },
});

export default ApolloServer;
