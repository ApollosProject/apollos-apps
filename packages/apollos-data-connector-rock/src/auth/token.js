import { AuthenticationError } from 'apollo-server';
import jwt from 'jsonwebtoken';

export const secret = process.env.SECRET || 'ASea$2gadj#asd0';

export const parseToken = (token) => jwt.verify(token, secret);

export const registerToken = (token) => {
  try {
    const { cookie, sessionId } = parseToken(token);

    return {
      userToken: token,
      rockCookie: cookie,
      sessionId,
    };
  } catch (e) {
    if (e instanceof jwt.TokenExpiredError) {
      return {};
    }
    throw new AuthenticationError('Invalid token');
  }
};

export const generateToken = (params) =>
  jwt.sign({ ...params }, secret, { expiresIn: '400d' });
