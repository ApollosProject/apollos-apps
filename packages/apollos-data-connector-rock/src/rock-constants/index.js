import RockApolloDataSource from '@apollosproject/rock-apollo-data-source';
import Config from '@apollosproject/config';

const { ROCK_MAPPINGS } = Config;

const mapApollosNameToRockName = (name) => {
  if (ROCK_MAPPINGS.CONTENT_ITEM_TYPES.includes(name)) {
    return 'ContentChannelItem';
  }
  throw new Error(`${name} has not been mapped into a Rock type!`);
};

class RockConstants extends RockApolloDataSource {
  async findOrCreate({ model, objectAttributes }) {
    // Turns {ChannelId: 7, Name: 'Something'} into '(ChannelId eq 7) and (Name eq 'Something')'
    const filter = Object.keys(objectAttributes)
      .map((key) => {
        if (typeof objectAttributes[key] === 'string') {
          return `(${key} eq '${objectAttributes[key]}')`;
        }
        return `(${key} eq ${objectAttributes[key]})`;
      })
      .join(' and ');

    const objects = await this.request(model)
      .filter(filter)
      .cache({ ttl: 86400 })
      .get();
    if (objects.length) {
      return objects[0];
    }
    const objectId = await this.post(`/${model}`, objectAttributes);
    const ret = await this.get(`/${model}/${objectId}`);
    console.log('Created', ret);
    return ret;
  }

  async createOrFindInteractionComponent({
    componentName,
    channelId,
    entityId,
  }) {
    return this.findOrCreate({
      model: 'InteractionComponents',
      objectAttributes: {
        Name: componentName,
        ChannelId: channelId,
        EntityId: entityId,
      },
    });
  }

  async createOrFindInteractionChannel({ channelName, entityTypeId }) {
    return this.findOrCreate({
      model: 'InteractionChannels',
      objectAttributes: {
        Name: channelName,
        UsesSession: true,
        IsActive: true,
        ComponentEntityTypeId: entityTypeId,
        ChannelTypeMediumValueId:
          ROCK_MAPPINGS.INTERACTIONS.CHANNEL_MEDIUM_TYPE_ID,
      },
    });
  }

  async contentItemInteractionComponent({ contentItemId, contentName = null }) {
    const channel = await this.contentItemInteractionChannel();
    return this.createOrFindInteractionComponent({
      componentName: `${
        ROCK_MAPPINGS.INTERACTIONS.COMPONENT_NAME
      } - ${contentName || contentItemId}`,
      channelId: channel.id,
      entityId: parseInt(contentItemId, 10),
    });
  }

  async contentItemInteractionChannel() {
    const { id } = await this.modelType('ContentItem');
    return this.createOrFindInteractionChannel({
      channelName: ROCK_MAPPINGS.INTERACTIONS.CHANNEL_NAME,
      entityTypeId: id,
    });
  }

  async modelType(nameInput) {
    const name = mapApollosNameToRockName(nameInput);

    const types = await this.request('EntityTypes')
      .filter(`Name eq 'Rock.Model.${name}'`)
      .cache({ ttl: 86400 })
      .get();
    if (types.length) {
      return types[0];
    }

    return null;
  }
}
// eslint-disable-next-line import/prefer-default-export
export { RockConstants as dataSource };
