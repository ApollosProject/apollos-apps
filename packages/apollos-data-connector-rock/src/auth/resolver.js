import { createGlobalId } from '@apollosproject/server-core';

const resolver = {
  Query: {
    currentUser: (root, args, { dataSources }) =>
      dataSources.Auth.getCurrentPerson(),
  },
  AuthenticatedUser: {
    id: ({ id }, args, context, { parentType }) =>
      createGlobalId(id, parentType.name),
    profile: (authUser) => authUser,
    rock: (rockDetails) => rockDetails,
    rockToken: (root, args, context) => context.rockCookie,
  },
  RockPersonDetails: {
    authToken: (root, args, { dataSources }) => dataSources.Auth.getAuthToken(),
    authCookie: (root, args, { rockCookie }) => rockCookie,
  },
  Authentication: {
    user: (root, args, { dataSources }) => dataSources.Auth.getCurrentPerson(),
  },
  Mutation: {
    authenticate: (root, { identity, password }, { dataSources }) =>
      dataSources.Auth.authenticate({ identity, password }),
    changePassword: (root, { password }, { dataSources }) =>
      dataSources.Auth.changePassword({ password }),
    registerPerson: (root, args, { dataSources }) =>
      dataSources.Auth.registerPerson(args),
  },
};

export default resolver;
