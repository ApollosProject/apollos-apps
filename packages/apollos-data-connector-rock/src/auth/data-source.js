import { AuthenticationError } from 'apollo-server';
import { fetch, Request } from 'apollo-server-env';
import moment from 'moment';
import RockApolloDataSource from '@apollosproject/rock-apollo-data-source';

import { generateToken, registerToken } from './token';

export default class AuthDataSource extends RockApolloDataSource {
  resource = 'Auth';

  rockCookie = null;

  userToken = null;

  initialize(config) {
    super.initialize(config);
    if (config.context.rockCookie) {
      // fetches the current person
      // this method will try to cache the current person on the context
      // removing the need to fetch each and every time
      this.getCurrentPerson();
    }
  }

  getCurrentPerson = async ({ cookie = null } = { cookie: null }) => {
    const { rockCookie, currentPerson } = this.context;
    const userCookie = cookie || rockCookie;

    if (currentPerson) {
      return currentPerson;
    }

    if (userCookie) {
      try {
        const request = await this.request('People/GetCurrentPerson').get({
          options: {
            headers: { cookie: userCookie, 'Authorization-Token': null },
          },
        });
        this.context.currentPerson = request;
        return request;
      } catch (e) {
        throw new AuthenticationError('Invalid user cookie');
      }
    }
    throw new AuthenticationError('Must be logged in');
  };

  fetchUserCookie = async (Username, Password) => {
    try {
      // We use `new Response` rather than string/options b/c if conforms more closely with ApolloRESTDataSource
      // (makes mocking in tests WAY easier to use `new Request` as an input in both places)
      const response = await fetch(
        new Request(`${this.baseURL}/Auth/Login`, {
          method: 'POST',
          body: JSON.stringify({
            Username,
            Password,
          }),
          headers: {
            'Content-Type': 'Application/Json',
          },
        })
      );
      if (response.status >= 400) throw new AuthenticationError();
      const cookie = response.headers.get('set-cookie');
      return cookie;
    } catch (err) {
      throw new AuthenticationError('Invalid Credentials');
    }
  };

  createSession = async ({ cookie }) => {
    const currentUser = await this.getCurrentPerson({ cookie });
    return this.post('/InteractionSessions', {
      PersonAliasId: currentUser.primaryAliasId,
    });
  };

  authenticate = async ({ identity, password }) => {
    try {
      const cookie = await this.fetchUserCookie(identity, password);
      const sessionId = await this.createSession({ cookie });
      const token = generateToken({ cookie, sessionId });
      const { userToken, rockCookie } = registerToken(token);
      this.context.rockCookie = rockCookie;
      this.context.userToken = userToken;
      this.context.sessionId = sessionId;
      return { token, rockCookie };
    } catch (e) {
      throw e;
    }
  };

  personExists = async ({ identity }) => {
    const hasUserName = await this.request(
      `/UserLogins?$filter=UserName eq '${identity}'`
    ).get();

    if (hasUserName.length) {
      return true;
    }
    return false;
  };

  createUserProfile = async (props = {}) => {
    try {
      const { email } = props;

      return await this.post('/People', {
        Email: email,
        IsSystem: false, // Required by Rock
        Gender: 0, // Required by Rock
      });
    } catch (err) {
      throw new Error('Unable to create profile!');
    }
  };

  createUserLogin = async (props = {}) => {
    try {
      const { email, password, personId } = props;

      return await this.post('/UserLogins', {
        PersonId: personId,
        EntityTypeId: 27, // A default setting we use in Rock-person-creation-flow
        UserName: email,
        PlainTextPassword: password,
        LastLoginDateTime: `${moment().toISOString()}`,
      });
    } catch (err) {
      throw new Error('Unable to create user login!');
    }
  };

  changePassword = async ({ password }) => {
    const currentUser = await this.getCurrentPerson();
    const { email, id } = currentUser;
    const login = await this.request('/UserLogins')
      .filter(`UserName eq '${email}'`)
      .first();

    if (login) {
      await this.delete(`/UserLogins/${login.id}`);
    }
    await this.createUserLogin({
      personId: id,
      email,
      password,
    });
    return this.authenticate({
      identity: email,
      password,
    });
  };

  registerPerson = async ({ email, password }) => {
    const personExists = await this.personExists({ identity: email });
    if (personExists) throw new Error('User already exists!');

    const personId = await this.createUserProfile({ email });
    await this.createUserLogin({
      email,
      password,
      personId,
    });

    const token = await this.authenticate({ identity: email, password });
    return token;
  };

  getAuthToken = async () => {
    const { rockCookie } = this.context;
    if (!rockCookie) throw new AuthenticationError('Must be logged in');

    // TODO remove this safety check and less secure implementation
    // once core Rock has the GetCurrentPersonImpersonationToken endpoint
    try {
      const token = await this.request(
        `People/GetCurrentPersonImpersonationToken?expireDateTime=${moment()
          .add(1, 'weeks')
          .toISOString()}`
      ).get({
        options: {
          headers: { cookie: rockCookie, 'Authorization-Token': null },
        },
      });
      return token;
    } catch (e) {
      console.warn(
        'Using deprecated Rock endpoint, upgrade Rock to v10 when available.'
      );
      const { id } = await this.getCurrentPerson();
      const param = await this.request(
        `People/GetImpersonationParameter?personId=${id}&expireDateTime=${moment()
          .add(1, 'weeks')
          .toISOString()}`
      ).get();
      return param.split('=')[1];
    }
  };
}
