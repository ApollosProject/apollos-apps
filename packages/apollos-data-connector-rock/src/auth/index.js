import { createGlobalId } from '@apollosproject/server-core';
import { get } from 'lodash';
import { registerToken } from './token';

export { registerToken, generateToken } from './token';
export { authSchema as schema } from '@apollosproject/data-schema';
export { default as dataSource } from './data-source';

export const contextMiddleware = ({ req, context: ctx }) => {
  if (get(req, 'headers.authorization')) {
    const { userToken, rockCookie, sessionId } = registerToken(
      req.headers.authorization
    );
    if (sessionId) {
      return {
        ...ctx,
        userToken,
        rockCookie,
        sessionId,
      };
    }
  }
  return ctx;
};

export const resolver = {
  Query: {
    currentUser: (root, args, { dataSources }) =>
      dataSources.Auth.getCurrentPerson(),
  },
  AuthenticatedUser: {
    id: ({ id }, args, context, { parentType }) =>
      createGlobalId(id, parentType.name),
    profile: (authUser) => authUser,
  },
  Authentication: {
    user: (root, args, { dataSources }) => dataSources.Auth.getCurrentPerson(),
  },
  Mutation: {
    authenticate: (root, { identity, password }, { dataSources }) =>
      dataSources.Auth.authenticate({ identity, password }),
    changePassword: (root, { password }, { dataSources }) =>
      dataSources.Auth.changePassword({ password }),
    registerPerson: (root, args, { dataSources }) =>
      dataSources.Auth.registerPerson(args),
  },
};
