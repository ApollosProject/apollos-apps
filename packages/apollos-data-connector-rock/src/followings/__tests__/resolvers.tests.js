import { graphql } from 'graphql';
import { fetch } from 'apollo-server-env';
import ApollosConfig from '@apollosproject/config';
import { createGlobalId } from '@apollosproject/server-core';
import { AuthenticationError } from 'apollo-server';
import { get } from 'lodash';

import {
  contentItemSchema,
  contentChannelSchema,
  themeSchema,
  scriptureSchema,
} from '@apollosproject/data-schema';
import { createTestHelpers } from '@apollosproject/server-core/lib/testUtils';
import { generateToken } from '../../auth';
import * as Followings from '../index';
import * as RockConstants from '../../rock-constants';

class ContentItemDataSource {
  // eslint-disable-next-line class-methods-use-this
  initialize() {}

  // eslint-disable-next-line class-methods-use-this
  getFromId(id) {
    return { id };
  }

  // eslint-disable-next-line class-methods-use-this
  getFromIds(ids) {
    return { get: () => ids.map((id) => ({ id })) };
  }
}

class AuthDataSource {
  initialize({ context }) {
    this.context = context;
  }

  getCurrentPerson() {
    if (this.context.currentPerson) {
      return { id: 'someId' };
    }
    throw new AuthenticationError('Must be logged in');
  }
}

const ContentItem = {
  dataSource: ContentItemDataSource,
  resolver: {
    ContentItem: {
      __resolveType: () => 'UniversalContentItem',
    },
    UniversalContentItem: {
      id: ({ id }) => createGlobalId(id, 'UniversalContentItem'),
    },
  },
};

const Auth = {
  dataSource: AuthDataSource,
  contextMiddleware: ({ req, context }) => {
    if (get(req, 'headers.authorization')) {
      return {
        ...context,
        currentPerson: true,
      };
    }
    return { ...context };
  },
};

const { getSchema, getContext } = createTestHelpers({
  Followings,
  RockConstants,
  ContentItem,
  UniversalContentItem: ContentItem,
  Auth,
});

ApollosConfig.loadJs({
  ROCK: {
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
  },
  ROCK_MAPPINGS: {
    SERIES_CONTENT_CHANNEL_TYPE_IDS: [6, 7],
    CONTENT_ITEM_TYPES: [
      'ContentItem',
      'UniversalContentItem',
      'DevotionalContentItem',
      'MediaContentItem',
    ],
  },
  APP: {
    DEEP_LINK_HOST: 'apolloschurch',
  },
});

describe('Following', () => {
  let schema;
  let context;
  beforeEach(() => {
    fetch.resetMocks();
    fetch.mockRockDataSourceAPI();
    schema = getSchema([
      contentItemSchema,
      contentChannelSchema,
      themeSchema,
      scriptureSchema,
    ]);
    const token = generateToken({ cookie: 'some-cookie', sessionId: 123 });
    context = getContext({
      req: {
        headers: { authorization: token },
      },
    });
  });

  it('likes an entity', async () => {
    const query = `
      mutation likeEntity {
        updateLikeEntity(
          input: {
            nodeId: "${createGlobalId(1, 'UniversalContentItem')}"
            operation: Like
          }
        ) {
          id
          isLiked
        }
      }
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });
  it('unlikes an entity', async () => {
    const query = `
      mutation likeEntity {
        updateLikeEntity(
          input: {
            nodeId: "${createGlobalId(1, 'UniversalContentItem')}"
            operation: Unlike
          }
        ) {
          id
          isLiked
        }
      }
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });
  it('uses following table to track if a user liked content', async () => {
    const query = `
      query getContent {
        node(id: "${createGlobalId(1, 'UniversalContentItem')}") {
          id
          ... on ContentItem {
            isLiked
          }
        }
      }
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('returns isLiked false if a user is logged out', async () => {
    const query = `
      query getContent {
        node(id: "${createGlobalId(1, 'UniversalContentItem')}") {
          id
          ... on ContentItem {
            isLiked
          }
        }
      }
    `;
    const contextWithoutUser = getContext();
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, contextWithoutUser);
    expect(result).toMatchSnapshot();
  });

  it('returns a likeCount', async () => {
    const query = `
      query getContent {
        node(id: "${createGlobalId(1, 'UniversalContentItem')}") {
          id
          ... on ContentItem {
            likedCount
          }
        }
      }
    `;
    const contextWithoutUser = getContext();
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, contextWithoutUser);
    expect(result).toMatchSnapshot();
  });

  it('gets all user liked content', async () => {
    const query = `
      query {
        getAllLikedContent {
          id
          isLiked
        }
      }
    `;
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('returns an empty array for liked content without a user logged in', async () => {
    const query = `
      query {
        getAllLikedContent {
          id
          isLiked
        }
      }
    `;
    const contextWithoutUser = getContext();
    const rootValue = {};
    const result = await graphql(schema, query, rootValue, contextWithoutUser);
    expect(result).toMatchSnapshot();
  });
});
