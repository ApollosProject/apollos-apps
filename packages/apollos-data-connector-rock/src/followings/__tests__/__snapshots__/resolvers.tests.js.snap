// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Following gets all user liked content 1`] = `
Object {
  "data": Object {
    "getAllLikedContent": Array [
      Object {
        "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
        "isLiked": true,
      },
    ],
  },
}
`;

exports[`Following likes an entity 1`] = `
Object {
  "data": Object {
    "updateLikeEntity": Object {
      "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "isLiked": true,
    },
  },
}
`;

exports[`Following returns a likeCount 1`] = `
Object {
  "data": Object {
    "node": Object {
      "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "likedCount": 1,
    },
  },
}
`;

exports[`Following returns an empty array for liked content without a user logged in 1`] = `
Object {
  "data": Object {
    "getAllLikedContent": Array [],
  },
}
`;

exports[`Following returns isLiked false if a user is logged out 1`] = `
Object {
  "data": Object {
    "node": Object {
      "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "isLiked": false,
    },
  },
}
`;

exports[`Following unlikes an entity 1`] = `
Object {
  "data": Object {
    "updateLikeEntity": Object {
      "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "isLiked": false,
    },
  },
}
`;

exports[`Following uses following table to track if a user liked content 1`] = `
Object {
  "data": Object {
    "node": Object {
      "id": "UniversalContentItem:559b23fd0aa90e81b1c023e72e230fa1",
      "isLiked": true,
    },
  },
}
`;
